warehouse/query-core/src/main/java/datawave/query/function/TLDEquality.java

```diff
@@ -23,8 +21,20 @@ public class TLDEquality implements Equality {
      */
     @Override
     public boolean partOf(Key key, Key other) {
-        ByteSequence docCF = TLD.estimateRootPointerFromId(key.getColumnFamilyData());
-        ByteSequence otherCF = TLD.estimateRootPointerFromId(other.getColumnFamilyData());
-        return otherCF.equals(docCF);
+        ByteSequence keyCf = key.getColumnFamilyData();
+        ByteSequence otherCf = other.getColumnFamilyData();
+
+        int dotCount = 0;
+        int len = Math.min(keyCf.length(), otherCf.length());
+        for (int i = 0; i < len; i++) {
+            byte a = keyCf.byteAt(i);
+            byte b = otherCf.byteAt(i);
+            if (a != b) {
+                return false;
+            } else if (a == '.' && ++dotCount == 3) {
```

Comment:
why 3?

Reply:
A root uid only has two dots, a child uid has three or more. Thus, if a third dot is encountered and everything up to that point has matched, then the root uids are the same.